NoClassDefFoundError: jakarta/servlet/ServletContext logged on Undertow startup

## Describe the bug
```
java.lang.NoClassDefFoundError: jakarta/servlet/ServletContext
	at co.elastic.apm.agent.servlet.servicename.ServletContainerInitializerServiceNameInstrumentation$JakartaInitServiceNameInstrumentation$AdviceClass.onEnter(ServletContainerInitializerServiceNameInstrumentation.java:109)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory$Initializer.onStartup(UndertowServletWebServerFactory.java:507)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:204)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:187)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:255)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory.createManager(UndertowServletWebServerFactory.java:330)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory.getWebServer(UndertowServletWebServerFactory.java:299)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:182)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:160)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:577)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:740)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:415)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1312)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1301)
	at uk.ac.csrf.dataplatform.DataPlatformApplication.main(DataPlatformApplication.java:16)
Caused by: java.lang.ClassNotFoundException: jakarta.servlet.ServletContext
	at net.bytebuddy.dynamic.loading.ByteArrayClassLoader.findClass(ByteArrayClassLoader.java:403)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at net.bytebuddy.dynamic.loading.ByteArrayClassLoader$ChildFirst.loadClass(ByteArrayClassLoader.java:1265)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 19 more
```

## Steps to reproduce
Spring Boot 2.6.5
Undertow 2.2.16
APM Agent 1.30.0
Run the application

## Expected behavior
No errors logged

--- a/CHANGELOG.asciidoc
+++ b/CHANGELOG.asciidoc
@@ -26,6 +26,8 @@ endif::[]
 [float]
 ===== Bug fixes
 * Fixed AWS Lambda instrumentation for AWS handler classes with input object types that are not AWS Events classes  - {pull}2551[#2551]
+* Fixed service name discovery based on MANIFEST.MF file through `ServletContainerInitializer#onStartup` on Jakarta Servlet containers -
+{pull}2546[#2546]
 
 [[release-notes-1.x]]
 === Java Agent version 1.x

--- a/apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/JakartaServletVersionInstrumentation.java
+++ b/apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/JakartaServletVersionInstrumentation.java
@@ -24,14 +24,9 @@ import net.bytebuddy.asm.Advice;
 
 import javax.annotation.Nullable;
 
-public abstract class JakartaServletVersionInstrumentation extends ServletVersionInstrumentation {
+public abstract class JakartaServletVersionInstrumentation {
 
-    @Override
-    public Constants.ServletImpl getImplConstants() {
-        return Constants.ServletImpl.JAKARTA;
-    }
-
-    public static class JakartaInit extends Init {
+    public static class JakartaInit extends ServletVersionInstrumentation.Init {
 
         @Override
         public Constants.ServletImpl getImplConstants() {
@@ -48,7 +43,7 @@ public abstract class JakartaServletVersionInstrumentation extends ServletVersio
 
     }
 
-    public static class JakartaService extends Service {
+    public static class JakartaService extends ServletVersionInstrumentation.Service {
 
         @Override
         public Constants.ServletImpl getImplConstants() {
@@ -61,7 +56,5 @@ public abstract class JakartaServletVersionInstrumentation extends ServletVersio
                 logServletVersion(JakartaUtil.getInfoFromServletContext(servlet.getServletConfig()));
             }
         }
-
     }
-
 }

--- a/apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/JavaxServletVersionInstrumentation.java
+++ b/apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/JavaxServletVersionInstrumentation.java
@@ -24,9 +24,9 @@ import javax.annotation.Nullable;
 import javax.servlet.Servlet;
 import javax.servlet.ServletConfig;
 
-public abstract class JavaxServletVersionInstrumentation extends ServletVersionInstrumentation {
+public abstract class JavaxServletVersionInstrumentation {
 
-    public static class JavaxInit extends Init {
+    public static class JavaxInit extends ServletVersionInstrumentation.Init {
 
         @Override
         public Constants.ServletImpl getImplConstants() {
@@ -43,7 +43,7 @@ public abstract class JavaxServletVersionInstrumentation extends ServletVersionI
 
     }
 
-    public static class JavaxService extends Service {
+    public static class JavaxService extends ServletVersionInstrumentation.Service {
 
         @Override
         public Constants.ServletImpl getImplConstants() {
@@ -56,7 +56,5 @@ public abstract class JavaxServletVersionInstrumentation extends ServletVersionI
                 logServletVersion(JavaxUtil.getInfoFromServletContext(servlet.getServletConfig()));
             }
         }
-
     }
-
 }

--- a/apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/servicename/ServletContainerInitializerServiceNameInstrumentation.java
+++ b/apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/servicename/ServletContainerInitializerServiceNameInstrumentation.java
@@ -97,7 +97,7 @@ public abstract class ServletContainerInitializerServiceNameInstrumentation exte
 
         @Override
         public Constants.ServletImpl getImplConstants() {
-            return Constants.ServletImpl.JAVAX;
+            return Constants.ServletImpl.JAKARTA;
         }
 
         public static class AdviceClass {

